### PR TITLE
Moved check for leech

### PR DESCRIPTION
**Pull request by Mrpox**
**Move the check for mana and life leech to make it not possible to leech life when just casting spells in the air without any target.
**Added new bool to check if already leeched for areacombat so it only leeches once when hitting multiple targets.

### CODE REVIEW

File: src/combat.cpp [hunk 1]
```diff
@@ -776,7 +775,9 @@ void Combat::doTargetCombat(Creature* caster, Creature* target, CombatDamage& da
 		if (g_game.combatBlockHit(damage, caster, target, params.blockedByShield, params.blockedByArmor, params.itemId != 0)) {
 			return;
 		}
-
+		if (casterPlayer) {
```
Comment (nekiro) [inline src/combat.cpp:778]: why did you move the code here?

File: src/combat.cpp [hunk 2]
```diff
@@ -905,6 +905,10 @@ void Combat::doAreaCombat(Creature* caster, const Position& position, const Area
 						if (g_game.combatBlockHit(damageCopy, caster, creature, params.blockedByShield, params.blockedByArmor, params.itemId != 0)) {
 							continue;
 						}
+						if (casterPlayer && !damage.leeched) {
```
Comment (nekiro) [inline src/combat.cpp:908]: same, why move there?
I agree with that bool though.

File: src/combat.cpp (hunk 1 above)
Comment (Mrpox) [reply src/combat.cpp:778]: I moved it here to be after block check.

File: src/combat.cpp (hunk 2 above)
Comment (Mrpox) [reply src/combat.cpp:908]: You're completely right, should check for highest damage and then leech.

File: src/combat.cpp (hunk 1 above)
Comment (Mrpox) [reply src/combat.cpp:778]: > why did you move the code here I think you can just check the target variable if it leechs even on no target?

I moved it here to be after block check.

File: src/combat.cpp (hunk 2 above)
Comment (Mrpox) [reply src/combat.cpp]: > For area spells this is actually still wrong. Cip calculates the highest damage from all the creatures that were hit in area spell and then calculates leech amount out of it. Although it should fix leeching for every target.

This has been altered, it now stores the highest damage done in areaattack and leeches off that.

**Comment by nekiro** [comment]
All the issues listen in this pr were fixed with https://github.com/otland/forgottenserver/pull/3267.
